### PR TITLE
[KYUUBI #1009][FOLLOWUP] Kyuubi server failed to start when connecting to secured HMS

### LINKED ISSUE #1161: [Bug] Kyuubi server failed to start when connecting to a secured Hive metastore
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

Kyuubi server failed to start when connecting to a secured Hive metastore

### Affects Version(s)

1.4.0

### Kyuubi Server Log Output

```logtalk
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/hadoop/hive/serde/serdeConstants
        at org.apache.hadoop.hive.metastore.MetaStoreUtils.<clinit>(MetaStoreUtils.java:830)
        at org.apache.hadoop.hive.metastore.RetryingMetaStoreClient.getProxy(RetryingMetaStoreClient.java:131)
        at org.apache.hadoop.hive.metastore.RetryingMetaStoreClient.getProxy(RetryingMetaStoreClient.java:89)
        at org.apache.kyuubi.credentials.HiveDelegationTokenProvider.initialize(HiveDelegationTokenProvider.scala:51)
        at org.apache.kyuubi.credentials.HadoopCredentialsManager.$anonfun$initialize$1(HadoopCredentialsManager.scala:96)
        at org.apache.kyuubi.credentials.HadoopCredentialsManager.$anonfun$initialize$1$adapted(HadoopCredentialsManager.scala:95)
        at scala.collection.immutable.Map$Map2.filterImpl(Map.scala:282)
        at scala.collection.immutable.Map$Map2.filterImpl(Map.scala:222)
        at scala.collection.TraversableLike.filter(TraversableLike.scala:395)
        at scala.collection.TraversableLike.filter$(TraversableLike.scala:395)
        at scala.collection.AbstractTraversable.filter(Traversable.scala:108)
        at org.apache.kyuubi.credentials.HadoopCredentialsManager.initialize(HadoopCredentialsManager.scala:95)
        at org.apache.kyuubi.service.CompositeService.$anonfun$initialize$1(CompositeService.scala:40)
        at org.apache.kyuubi.service.CompositeService.$anonfun$initialize$1$adapted(CompositeService.scala:40)
        at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
        at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
        at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
        at org.apache.kyuubi.service.CompositeService.initialize(CompositeService.scala:40)
        at org.apache.kyuubi.session.SessionManager.initialize(SessionManager.scala:165)
        at org.apache.kyuubi.session.KyuubiSessionManager.initialize(KyuubiSessionManager.scala:39)
        at org.apache.kyuubi.service.CompositeService.$anonfun$initialize$1(CompositeService.scala:40)
        at org.apache.kyuubi.service.CompositeService.$anonfun$initialize$1$adapted(CompositeService.scala:40)
        at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
        at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
        at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
        at org.apache.kyuubi.service.CompositeService.initialize(CompositeService.scala:40)
        at org.apache.kyuubi.service.AbstractBackendService.initialize(AbstractBackendService.scala:169)
        at org.apache.kyuubi.service.CompositeService.$anonfun$initialize$1(CompositeService.scala:40)
        at org.apache.kyuubi.service.CompositeService.$anonfun$initialize$1$adapted(CompositeService.scala:40)
        at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
        at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
        at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
        at org.apache.kyuubi.service.CompositeService.initialize(CompositeService.scala:40)
        at org.apache.kyuubi.service.Serverable.initialize(Serverable.scala:46)
        at org.apache.kyuubi.server.KyuubiServer.initialize(KyuubiServer.scala:100)
        at org.apache.kyuubi.server.KyuubiServer$.startServer(KyuubiServer.scala:46)
        at org.apache.kyuubi.server.KyuubiServer$.main(KyuubiServer.scala:77)
        at org.apache.kyuubi.server.KyuubiServer.main(KyuubiServer.scala)
Caused by: java.lang.ClassNotFoundException: org.apache.hadoop.hive.serde.serdeConstants
        at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
        ... 38 more
```

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

```yaml
`hive-site.xml` under kyuubi conf dir:


<configuration>
  <property>
    <name>hive.metastore.uris</name>
    <value>thrift://kdc.hadoop.com:9083</value>
  </property>
  <property>
    <name>hive.metastore.sasl.enabled</name>
    <value>true</value>
  </property>
  <property>
    <name>hive.metastore.kerberos.principal</name>
    <value>hive/_HOST@HADOOP.COM</value>
  </property>
</configuration>
```

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by zhouyifan279**
### _Why are the changes needed?_
Fix #1161 [Bug] Kyuubi server failed to start when connecting to a secured Hive metastore

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate
image

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/pom.xml [hunk 1]
```diff
@@ -76,11 +76,21 @@
             </exclusions>
         </dependency>
 
+        <dependency>
```
Comment (yaooqinn) [inline kyuubi-server/pom.xml:79]: can we test with hive metastore server 1.2

**Comment by zhouyifan279** [comment]
@pan3793 ready for review

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
Since we introduced hive deps in `kyuubi-server` module, I'm thinking to move all hive things of `kyuubi-hive-jdbc` from `pom.xml` into `kyuubi-hive-jdbc/pom.xml`. Then it doesn't bother us, and we can use different hive versions for `kyuubi-hive-jdbc` and `kyuubi-server`.

**Review by pan3793 — APPROVED** [review]
LGTM

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**